magento/magento2
# 12386: Order Status resets to default Status after Partial Refund.

### Original Pull Request
https://github.com/magento-engcom/magento2ce/pull/1056

### Description
Now order keeps previous status after partial refund, in case it's state has not been changed.

### Fixed Issues (if relevant)

1. magento/magento2#12386: Order Status resets to default Status after Partial Refund

### Manual testing scenarios

1. Create new Order Status: "Picking"
2. Assign new State to Processing: "processing[Picking]", Do not check "Use Order Status As Default".
3. Create simple product.
4. Create order and Invoice with created product (qty = 2).
5. Update order status from Processing to Picking.
6. Refund part of order whilst in Picking Status(create Credit Memo with qty = 1).
7. Observe order status remains "Picking".
8. Refund remaining order items.
9. Observe order status changes to "Closed".
10. Repeat steps 4 and 5.
11. Make API rest call POST /V1/order/{your_order_Id}/refund with body ex:
```
{
  "items": [
    {
      "order_item_id": {your_order_item_id},
      "qty": 1
    }
  ]
}
```
12. Check order status in admin, it should remains "Picking".
13. Repeat API call.
14. Check order status become "Closed".


### Contribution checklist
